src/main/scala/xiangshan/backend/CtrlBlock.scala

```diff
@@ -323,13 +332,21 @@ class CtrlBlockImp(
   redirectGen.io.oldestExuOutPredecode.valid := GatedValidRegNext(oldestExuPredecode.valid)
   redirectGen.io.oldestExuOutPredecode := RegEnable(oldestExuPredecode, oldestExuPredecode.valid)
   redirectGen.io.loadReplay <> loadReplay
-  val loadRedirectOffset = Mux(memViolation.bits.flushItself(), 0.U, Mux(memViolation.bits.isRVC, 2.U, 4.U))
-  val loadRedirectPcFtqOffset = RegEnable((memViolation.bits.ftqOffset << instOffsetBits).asUInt +& loadRedirectOffset, memViolation.valid)
-  val loadRedirectPcRead = pcMem.io.rdata(pcMemRdIndexes("redirect").head).toUInt + loadRedirectPcFtqOffset
-
-  redirectGen.io.loadReplay.bits.pc := loadRedirectPcRead
-  val load_target = loadRedirectPcRead
-  redirectGen.io.loadReplay.bits.target := load_target
+  val loadRedirectTargetOffset = Reg(UInt(VAddrBits.W))
+  when(memViolation.valid) {
+    val thisPcOffset = memViolation.bits.getPcOffset()
+    val nextPcOffset = memViolation.bits.getNextPcOffset()
+    loadRedirectTargetOffset := Mux(memViolation.bits.flushItself(), thisPcOffset, nextPcOffset)
+  }
+  val loadRedirectStartPcRead = pcMem.io.rdata(pcMemRdIndexes("redirect").head).toUInt
+  val load_target = loadRedirectStartPcRead + loadRedirectTargetOffset
+  redirectGen.io.loadReplay.bits.cfiUpdate.target := load_target
+  // TODO loadRedirectPcOffset is useful?
+  val loadRedirectPcOffset = Reg(UInt(VAddrBits.W))
+  when(memViolation.valid) {
+    loadRedirectPcOffset := memViolation.bits.getPcOffset()
+  }
+  redirectGen.io.loadReplay.bits.cfiUpdate.pc := loadRedirectStartPcRead + loadRedirectPcOffset
```

Comment:
```suggestion
  redirectGen.io.loadReplay.bits.target := load_target
  // TODO loadRedirectPcOffset is useful?
  val loadRedirectPcOffset = Reg(UInt(VAddrBits.W))
  when(memViolation.valid) {
    loadRedirectPcOffset := memViolation.bits.getPcOffset()
  }
  redirectGen.io.loadReplay.bits.pc := loadRedirectStartPcRead + loadRedirectPcOffset
```

`CfiUpdateInfo` is removed since #4945